web/src/beta/components/Modal/index.tsx

```diff
@@ -1,54 +1,163 @@
   className?: string;
-  title?: string;
-  size?: "sm" | "md" | "lg";
+  modalTitle?: string;
+  size?: Size;
   button1?: ReactNode;
   button2?: ReactNode;
   children?: ReactNode;
   isVisible?: boolean;
   onClose?: () => void;
+  showSidebar?: boolean;
+  sidebarTabs?: SidebarTab[];
 };
 
 const Modal: React.FC<Props> = ({
   className,
-  title,
-  size,
+  modalTitle,
+  size = "md",
   button1,
   button2,
+  children,
   isVisible,
   onClose,
-  children,
+  sidebarTabs = [],
 }) => {
-  const theme = useTheme();
+  const [selectedTab, setSelectedTab] = useState<string | null>(
+    sidebarTabs.length > 0 ? sidebarTabs[0].id : null,
+  );
+
+  const handleTabChange = (tabId: string) => {
+    setSelectedTab(tabId);
+  };
+
   return (
-    <Wrapper className={className} size={size} isVisible={isVisible} onClose={onClose}>
-      <Title size="h4" weight="bold" color={theme.general.content.strong}>
-        {title}
-      </Title>
-      {children}
-      <ButtonWrapper>
-        {button2}
-        {button1}
-      </ButtonWrapper>
+    <Wrapper
+      className={className}
+      size={size}
+      isVisible={isVisible}
+      modalTitle={modalTitle}
+      onClose={onClose}>
+      {sidebarTabs.length > 0 ? (
+        <SidebarWrapper>
+          <NavBarWrapper>
+            {sidebarTabs.map(tab => (
+              <Tab
+                key={tab.id}
+                isSelected={selectedTab === tab.id}
+                onClick={() => handleTabChange(tab.id)}>
+                {tab.label}
+              </Tab>
+            ))}
+          </NavBarWrapper>
+          <SidebarContentWrapper>
+            <TabContent>
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.content}
+            </TabContent>
+            <TabButtonWrapper>
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.tabButton1}
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.tabButton2}
+            </TabButtonWrapper>
+          </SidebarContentWrapper>
+        </SidebarWrapper>
+      ) : (
+        <>
+          <ContentWrapper>{children}</ContentWrapper>
+          <ButtonWrapper>
+            {button2}
+            {button1}
+          </ButtonWrapper>
+        </>
+      )}
     </Wrapper>
   );
 };
 
-const Title = styled(Text)`
-  text-align: center;
-  margin-bottom: 24px;
+const SidebarWrapper = styled.div`
+  display: flex;
+  align-items: flex-start;
+  align-self: stretch;
+`;
+
+const NavBarWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 16px;
+  align-items: flex-start;
+  gap: 10px;
+  align-self: stretch;
+  border-right: 1px solid ${({ theme }) => theme.editor.weakOutline};
+`;
+
+const Tab = styled.button<{ isSelected: boolean }>`
+  display: flex;
+  padding: 4px 8px;
+  align-items: flex-start;
+  align-self: stretch;
+  border-radius: 4px;
+  background: ${({ isSelected, theme }) => (isSelected ? theme.editor.bg1 : `transparent`)};
```

Comment:
Please only use back ticks if you are going to embed javascript values in to the string. Stick to normal "" instead

Reply:
done